src/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs

```diff
@@ -94,5 +97,38 @@ public long ExtendedExpiresIn
 
         public DateTimeOffset AccessTokenExpiresOn { get; private set; }
         public DateTimeOffset AccessTokenExtendedExpiresOn { get; private set; }
+
+        public string Authority { get; private set; }
+
+        internal MsalTokenResponse CreateFromBrokerResponse(Dictionary<string, string> responseDictionary)
```

Comment:
"builder" methods are typically static, because they act as constructors. You should not need to have an instance of MsalTokenResponse to create another instance of MsalTokenResponse